### ISSUE TITLE
Could not resolve module/action even though it exists

### ISSUE THREAD

**Issue by g-gill1995** [aruba/aos-switch-ansible-collection#4]
---
- hosts: all
  gather_facts: false
  connection: local
  become: yes
  become_method: enable

  collections:
    - arubanetworks.aos_switch

  tasks:
    - name: Show running configuration
      arubanetworks.aos_switch.arubaoss_command:
        commands: ['show run']

When I run in AWX:

ERROR:
ERROR! couldn't resolve module/action 'arubanetworks.aos_switch.arubaoss_command'

I have check ~/.ansible/collections/ansible_collections

I can see arubanetworks/aos_switch/plugins/modules/aubaoss_command.py BUT I still keep getting this error. Making me crazy.

ansible-galaxy collection list does show arubanetworks

**Comment by tchiapuziowong** [comment]
Hi @g-gill1995 thanks for this! There are a couple things to note.

1) When you use the modules inside the playbook you don't preface them with `arubanetwork.aos_switch` so your task should be this:
```yaml
    - name: Show running configuration
      arubaoss_command:
        commands: ['show run']
```
2) For your Ansible connection it should be set to `ansible_connection: network_cli` if you're wanting to use the SSH CLI modules which is what you're using in this case. For more info on how to use these modules and the REST API modules see [Using Both REST API and SSH/CLI Modules on a Host](https://github.com/aruba/aos-switch-ansible-collection#using-both-rest-api-and-sshcli-modules-on-a-host). 
Here is what the final playbook should look like:
```yaml
- hosts: all
  gather_facts: false
  vars:
    ansible_connection: network_cli
  become: yes
  become_method: enable
  collections:
    - arubanetworks.aos_switch
  tasks:
    - name: Show running configuration
      arubaoss_command:
        commands: ['show run']
```

Please let me know if this resolves your issue!

**Comment by g-gill1995** [comment]
Hey,

I did have my config like that below but it still wasn’t working. Let me try that one again!

**Comment by tchiapuziowong** [comment]
When you execute your playbook you also need to have `arubaoss` added as a part of your `NETWORK_GROUP_MODULES`. See [Setting Environment Variables](https://github.com/aruba/aos-switch-ansible-collection#setting-environment-variables) for how to do this in your `ansible.cfg` file or on the command line!

**Comment by g-gill1995** [comment]
Hey,

Thanks for your assistance but I still seem to be getting the same error
This is the playbook: https://github.com/g-gill1995/aruba-switch-ansible

[cid:image001.png@01D6ABA6.F52D0EE0]
Collection:

All the collections seem to work except the arubanetworks.aos_switch ☹

[cid:image002.png@01D6ABA6.F52D0EE0]

**Comment by tchiapuziowong** [comment]
Can you copy/paste the CLI command you're executing with the error? Also can you copy/paste the output of this command? `ansible-galaxy collection list`

**Comment by g-gill1995** [comment]
Hey,

Are you able to see the screenshots? I posted some in the last reply.

**Comment by tchiapuziowong** [comment]
No sorry I'm not seeing any, there are no pictures in the Github issue :(

**Comment by g-gill1995** [comment]
Ahh ok one sec, let me try again!

**Comment by g-gill1995** [comment]
![image](https://user-images.githubusercontent.com/72466294/97216451-42eb8380-179c-11eb-8971-d07cc9e98376.png)

![image](https://user-images.githubusercontent.com/72466294/97216518-572f8080-179c-11eb-8c39-b7c3b2cf8fbe.png)

**Comment by g-gill1995** [comment]
Added!

**Comment by tchiapuziowong** [comment]
Okay it seems like you're running your playbook using Ansible Tower! When you install collections you have to ensure you're installing them in the correct path that Tower is searching for! Try installing the collection with the following command and see if that fixes your issue:
`ansible-galaxy collection install arubanetworks.aoscx -p /usr/share/ansible/collections`

For more information: https://developer.arubanetworks.com/aruba-aoscx/docs/ansible-tower-overview

**Comment by g-gill1995** [comment]
Hey,

Installed it in that directory and im still getting the same error :(!!

SSH password: 
ERROR! couldn't resolve module/action 'arubaoss_command'. This often indicates a misspelling, missing collection, or incorrect module path.
The error appears to be in '/tmp/awx_430_bkthrumn/project/Show Run.yml': line 11, column 7, but may
be elsewhere in the file depending on the exact syntax problem.
The offending line appears to be:
  tasks:
    - name: Show running configuration
      ^ here

It's weird because my cisco asa playbook is under ~/.ansible/collections/ansible_collections and not /usr/share/ansible/collections but im able to run cisc asa playbooks just fine through awx :o

**Comment by tchiapuziowong** [comment]
Oh you're using AWX and not Tower? Is the collection installed within the AWX container?

**Comment by g-gill1995** [comment]
Hey,

I have it under ~/.ansible/collections/ansible_collections with my other collections. Any tips on how to try and install it within the AWX container (sorry new to this whole AWX/Ansible)

Thank You!

**Comment by tchiapuziowong** [comment]
I can't find any specific Ansible documentation that describes the process but I typically just enter the awx_task container and execute the ansible-galaxy install command

**Comment by craigers521** [comment]
Gil - im also running AWX in a container.  All i had to do was create a requirements.yml file and my repo inside a directory named "collections" (so the path from the root of the repo is collections/requirements.yml) and include all the collections i want to install (not AWX looks specifically for the file in this directory so it has to be .yml and not .yaml).  The requirements file i have looks like this: 

```
---
collections:
- netbox.netbox
- arubanetworks.aos_switch
- cisco.ios
```

When i run the job to sync my project from the repo it automatically will install the collections. 

![image](https://user-images.githubusercontent.com/15713242/97316249-87782d00-1837-11eb-86a9-d28118f28493.png)


I havent upgraded AWX to use 2.10 yet (they dont have this dockerized quite yet) but i plan on doing that this week and testing the new collection inside AWX... i can let you know

**Comment by g-gill1995** [comment]
> Gil - im also running AWX in a container. All i had to do was create a requirements.yml file and my repo inside a directory named "collections" (so the path from the root of the repo is collections/requirements.yml) and include all the collections i want to install (not AWX looks specifically for the file in this directory so it has to be .yml and not .yaml). The requirements file i have looks like this:
> 
> ```
> ---
> collections:
> - netbox.netbox
> - arubanetworks.aos_switch
> - cisco.ios
> ```
> 
> When i run the job to sync my project from the repo it automatically will install the collections.
> 
> ![image](https://user-images.githubusercontent.com/15713242/97316249-87782d00-1837-11eb-86a9-d28118f28493.png)
> 
> I havent upgraded AWX to use 2.10 yet (they dont have this dockerized quite yet) but i plan on doing that this week and testing the new collection inside AWX... i can let you know

That workedd

Thank You!!

Btw what do you have for ansible_network_os ? When I used arubaoss it tells me its not found.

**Comment by g-gill1995** [comment]
nvm fixed it with ansible_network_os: arubanetworks.aos_switch.arubaoss

You are a life saver man!

**Comment by khokarv** [comment]
Issue: 

ERROR! couldn't resolve module/action 'f5networks.f5_modules.bigip_selfip'. This often indicates a misspelling, missing collection, or incorrect module path.

Resolution: 

Step 1: Fixed path in ansible config file:
[root@tower-uae2 ~]# grep -i collection /etc/ansible/ansible.cfg
collections_paths = /var/lib/awx/vendor/inventory_collections/ansible_collections
[root@tower ~]#

Step 2: Install module in correct path:
[root@tower ~]# ansible-galaxy collection install f5networks.f5_modules -p /usr/share/ansible/collections
[WARNING]: The specified collections path '/usr/share/ansible/collections' is not part of the configured Ansible collections paths '/var/lib/awx/vendor/inventory_collections/ansible_collections'. The installed collection won't be
picked up in an Ansible run.
Process install dependency map
\
/
Starting collection install process
Installing 'f5networks.f5_modules:1.15.0' to '/usr/share/ansible/collections/ansible_collections/f5networks/f5_modules'
Installing 'ansible.netcommon:2.6.1' to '/usr/share/ansible/collections/ansible_collections/ansible/netcommon'
Installing 'ansible.utils:2.5.2' to '/usr/share/ansible/collections/ansible_collections/ansible/utils'
[root@tower-uae2 ~]#

**Comment by foxpatil** [comment]
@khokarv @tchiapuziowong @g-gill1995 @martinl 

We have ansible tower and installed below galaxy collection

ansible-galaxy collection install fortinet.fortios -p /usr/share/ansible/collections

but it is still giving below error:

ERROR! couldn't resolve module/action 'fortios_monitor_fact'. This often indicates a misspelling, missing collection, or incorrect module path.

fortinet.fortios is mentioned under collection

below is my playbook
---

- hosts: fortigate,syslog
  connection: httpapi
  gather_facts: false
  
  collections:
   - fortinet.fortios
 
tasks:
         
 
        -   fortios_monitor_fact:           
                  selector: "system_config_backup"
                  vdom: global
                  
                  params:
                    scope: global

            register: backup

**Comment by tchiapuziowong** [comment]
@foxpatil I'm sorry to hear you're having difficulty but this is not a HPE Aruba Networking supported collection - please refer to the owners of `fortios` for troubleshotting

**Comment by foxpatil** [comment]
@tchiapuziowong Thank you for quick repose.. but this will be the correct way to install any of the collection and use under playbook..

Ansible tower tries which path to execute playbook ?

**Comment by tchiapuziowong** [comment]
@foxpatil  are you using Ansible Tower or the Ansible Automation Platform? If you're paying for support through Red Hat perhaps they can assist you in troubleshooting?

Here are some guides that discuss collections path:
